tendrl/notifier/notification/__init__.py

```diff
@@ -122,9 +123,21 @@ def run(self):
                 for alert in alerts:
                     alert.tags = json.loads(alert.tags)
                     if alert.delivered == "False":
+                        lock = etcd.Lock(
+                            NS._int.wclient,
+                            'alerting/alerts/%s' % alert.alert_id
+                        )
+                        lock.acquire(
+                            blocking=True,
+                            lock_ttl=60
+                        )
+                    if lock.is_acquired:
```

Comment:
this if condition is outside the scope in which the "lock" is created and acquired,

youll need to move it inside the scope, current code will fail if alert.delivered != "False"

Reply:
yes, i need to move why it is not failed and works fine means, an exception is caught by exception handler and for it worked for undelivered alerts fine. i will change this